### DISCUSSION TITLE
Development: Change status lable workflow

### DISCUSSION THREAD

**Discussion by capoaira** [2Abendsegler/GClh#2430] [Ideas]
Ich habe es vielleicht schonmal angesprochen. Meiner Meinung nach ist der Workflow mit den Labels unnötig aufwendig. Zudem müssen wir uns mittlerweile durch eine recht lange Liste durch scrollen.

Dabei bietet GitHub einige Labels mehr oder weniger direkt selbst, wenn man die Keywords zum Schießen von PRs (fixes #Nr oder close #Nr oder...) nutzt.

Im Moment müssen wir bei jedem Issue labels setzten:

1. `In Progess` (gleichzeitig assignen wir uns)
2. `waiting for merge` (gleichzeitig erstellen wir eine PR)
3. `Complete` / `fixed` (gleichzeitig schießen wir das Issee)

Dann müssen wir nach dem merge das Issue von Hand schließen (und wir fügen noch hinzu, mit welcher PR)

Eigentlich kann man die Schritte jedoch auch so erkennen
Jemand hat sich assigned: `In Progess`
Eine PR wurde erstellt: `waiting for merge`
Die PR wurde gemerged: `Complete` / `fixed`

Durch die Keyword-Automatik von GitHub wird automatisch darauf hingewiesen, dass eine Pull Request für das Issue besteht und beim automatischen Schlissen wird die PR auch automatisch erwähnt.

Auch das Label `answered` benötigen wir eigentlich nicht, da Github Discussions das ja auch mitliefert.

**Comment by capoaira** [comment]
Dafür sollten wir vielleicht ein Label "Due Chages by GS" oder so einführen, welches wir setzten, wenn etwas nicht funktioniert, weil mal wieder etwas geändert wurde 😅

**Comment by 2Abendsegler** [comment]
@capoaira 
Hey, 
deine Discussion ist an mir vorbei gegangen, ich habe keine Info darüber erhalten. Habs gerade nur durch Zufall gesehen.
Tut mir leid.
Ich überlege mir mal was wir von deinen Vorschlägen umsetzen können. 
Kannst mich das nächste Mal wenn ich mich nicht melde ruhig anpingen. 😊
Schönes WE

**Comment by 2Abendsegler** [comment]
@capoaira 
Moin,
ich hatte unbeabsichtigt nicht alle Mitteilungen abonniert (Only receive notifications from this repository when participating or @mentioned.). Womöglich ist auch an anderer Stelle etwas verloren gegangen. Falls du was im Gedächtnis hast, sag bitte Bescheid.

**Comment by 2Abendsegler** [comment]
@capoaira 
Eigentlich habe ich mir nie wirklich Gedanken um den Workflow gemacht, weil die Anzahl an neuen Issues und PRs doch ziemlich überschaubar ist. Der heutige Zustand ist einfach im Laufe der Zeit gewachsen. Ich muss zugeben, dass ich die grünen Status Labels ziemlich mag, weil ich blitzschnell in einer Liste sehen kann, wenn etwas aus dem Rahmen fällt weil es kein grünes Label hat. 

Wir können aber trotzdem gerne Vereinfachungen vornehmen. Das geht aber aus meiner Sicht immer oder zumindest fast immer zu Lasten des Informationsgehaltes.

---
> 1. In Progess (gleichzeitig assignen wir uns)  

Das Assignen eines Issues bedeutet nicht zwangsläufig in progress. Es kann auch bedeuten, dass der Assignte das Issue gerne bearbeiten möchte. So war es beispielsweise beim [Log Form] GClh header disappears. #2443. Ich hatte eine genaue Vorstellung davon wie ich den Bug reproduzieren konnte, hatte aber noch gar keine Zeit daran zu arbeiten. Wenn also jemand anderes das Issue auch hätte bearbeiten wollen, dann wäre das noch ein guter Zeitpunkt. Sobald ich den Status in progress gesetzt habe, bin ich echt daran am arbeiten.

So wichtig ist das aber nicht, das können wir trotzdem weglassen. 

> 2. waiting for merge (gleichzeitig erstellen wir eine PR)

In der Liste der Issues kann ich nicht erkennen, ob es einen PR für das Issue gibt und ob er Draft, Open, Closed oder Merged ist und ob ein Review requested ist. Ohne Status Label kann ich in dieser Liste nicht erkennen welchen Status das Issue hat. 

Da wir nicht so viele Issues haben, ist das aber nicht so wichtig. Das können wir also auch weglassen.

> Dafür sollten wir vielleicht ein Label "Due Chages by GS" oder so einführen, welches wir setzten, wenn etwas nicht funktioniert, weil mal wieder etwas geändert wurde ??

Ja, hatte ich auch mal dran gedacht. Eigentlich würde so etwas womöglich aber vermitteln, dass GS hier Schuld hat, was ja aber nicht der Fall ist. Die haben nur etwas geändert an ihren eigenen Webseiten. Das ist ja völlig in Ordnung. Fehler die dadurch bei uns entstehen, können wir eigentlich nicht GS anlasten. Ich hatte früher immer mal im Titel des Issues oder PRs "due to changes by GS" oder so ähnlich geschrieben. Mach ich aber aus beschriebener Überlegung heute nicht mehr. 

---
Vorschlag fur die zukünftige minimale Vorgehensweise, mehr kann man aber immer tun, auch im Nachhinein: 

- Die Labels bug, enhancement, improvement ... sollten weiterhin in Issues und eigenständigen PRs verwendet werden. 
- Die grünen Status Labels in progress, waiting for merge, completed, fixed, answered oder rejected sind nicht nötig, dürfen aber weiterhin verwendet werden.
- Um einen PR zum Review zu geben, oder nach erfolgtem Review erneut zum Review zu geben, reicht ein Open PR und ein Request Reviewer. Ein Kommentar ist nicht notwendig, es sei denn es gilt beim Review etwas besonderes zu beachten. 
- Erfolgt beim Review kein Approval, sollte der PR wieder auf Draft gesetzt werden.
- Ein Issue darf mit dem Merge des PR automatisch geschlossen werden. Ein Kommentar im Issue ist nicht notwendig, es sei denn eine weitere Mitteilung an den Ersteller des Issues ist geboten.
- Ein Issue darf manuell mit "Close as completed" bzw. "Close as not planned" geschlossen werden. Ein Kommentar im Issue ist nicht notwendig, es sei denn eine weitere Mitteilung an den Ersteller des Issues ist geboten.

**Reply by capoaira** [reply]
Ja, das hört sich gut an. 👍 Wir können ja auch immer noch zum Alten zurückkehren, wenn wir merken, dass es so zu unübersichtlich ist.

>Eigentlich würde so etwas womöglich aber vermitteln, dass GS hier Schuld hat, was ja aber nicht der Fall ist. Die haben nur etwas geändert an ihren eigenen Webseiten. Das ist ja völlig in Ordnung. Fehler die dadurch bei uns entstehen, können wir eigentlich nicht GS anlasten. 

Das stimmt natürlich, soweit habe ich gar nicht gedacht 😅
